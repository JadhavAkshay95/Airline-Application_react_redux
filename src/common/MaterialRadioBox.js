import React from 'react'
import Radio from '@material-ui/core/Radio'
import RadioGroup from '@material-ui/core/RadioGroup'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import FormControl from '@material-ui/core/FormControl'
import PropTypes from 'prop-types'

const MaterialRadioBox = ({
  radioGroupName,
  options,
  value,
  handleRadioChange
}) => {
  return (
    <>
      <FormControl component="fieldset">
        <RadioGroup
          aria-label={radioGroupName}
          name={radioGroupName}
          value={value}
          onChange={handleRadioChange}
        >
          {options.map(option => (
            <FormControlLabel
              key={option.value}
              value={option.value}
              control={<Radio />}
              label={option.label}
            />
          ))}
        </RadioGroup>
      </FormControl>
    </>
  )
}

export default MaterialRadioBox
MaterialRadioBox.propTypes = {
  radioGroupName: PropTypes.string,
  options: PropTypes.array,
  value: PropTypes.string,
  handleRadioChange: PropTypes.func
}
