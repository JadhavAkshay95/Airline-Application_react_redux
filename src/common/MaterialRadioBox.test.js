import React from 'react'
import { shallow } from 'enzyme'
import MaterialRadioBox from './MaterialRadioBox'

const renderMaterialRadioBox = args => {
  const defaultProps = {
    radioGroupName: null,
    options: [],
    value: null,
    handleRadioChange: () => {}
  }
  const props = { ...defaultProps, ...args }
  return shallow(<MaterialRadioBox {...props} />)
}

describe('Render common material radio box', () => {
  it('Render common radio box component', () => {
    const wrapper = renderMaterialRadioBox({
      radioGroupName: 'foodPref',
      options: [
        { label: 'vegan', value: 'vegan' },
        { label: 'veg', value: 'veg' }
      ],
      value: 'vegan'
    })
    expect(wrapper).toMatchSnapshot()
  })
})
