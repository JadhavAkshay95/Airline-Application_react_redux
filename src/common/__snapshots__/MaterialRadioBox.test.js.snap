// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render common material radio box Render common radio box component 1`] = `ShallowWrapper {}`;
